sunpy/database/tables.py

```diff
@@ -339,8 +339,8 @@ def _from_query_result_block(cls, qr_block, default_waveunit=None):
         provider = str(
             qr_block.provider) if qr_block.provider is not None else None
         fileid = str(qr_block.fileid) if qr_block.fileid is not None else None
-        instrument = str(
-            qr_block.instrument) if qr_block.instrument is not None else None
+        instrument = str(qr_block.instrument) if qr_block.instrument is not None else None
+        size = qr_block.size if hasattr(qr_block, 'size') else -1
```

Comment:
I made the changes to other variables there as well. As for the `-1`, I don't know what it means because that line came from me fixing a merge conflict. It is already there in sunpy-master, and it was not introduced in this PR.